specification/migrate/resource-manager/Microsoft.Migrate/stable/2019-05-01/migrate.json

```diff
@@ -0,0 +1,3777 @@
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "security": [
+    {
+      "azure_auth": [
+        "user_impersonation"
+      ]
+    }
+  ],
+  "definitions": {
+    "Project": {
+      "type": "object",
+      "x-ms-azure-resource": true,
+      "description": "Azure Migrate Project.",
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Path reference to this project /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Migrate/assessmentProjects/{projectName}",
+          "readOnly": true
+        },
+        "name": {
+          "type": "string",
+          "description": "Name of the project.",
+          "readOnly": true
+        },
+        "type": {
+          "type": "string",
+          "description": "Type of the object = [Microsoft.Migrate/assessmentProjects].",
+          "readOnly": true
+        },
+        "eTag": {
+          "type": "string",
+          "description": "For optimistic concurrency control."
+        },
+        "location": {
+          "type": "string",
+          "description": "Azure location in which project is created."
+        },
+        "tags": {
+          "type": "object",
+          "description": "Tags provided by Azure Tagging service."
+        },
+        "properties": {
+          "description": "Properties of the project.",
+          "$ref": "#/definitions/ProjectProperties"
+        }
+      }
+    },
+    "ProjectProperties": {
+      "type": "object",
+      "description": "Properties of a project.",
+      "properties": {
+        "createdTimestamp": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Time when this project was created. Date-Time represented in ISO-8601 format.",
+          "readOnly": true
+        },
+        "updatedTimestamp": {
+          "type": "string",
+          "format": "date-time",
+          "description": "Time when this project was last updated. Date-Time represented in ISO-8601 format.",
+          "readOnly": true
+        },
+        "serviceEndpoint": {
+          "type": "string",
+          "description": "Endpoint at which the collector agent can call agent REST API.",
+          "readOnly": true
+        },
+        "assessmentSolutionId": {
+          "type": "string",
+          "description": "Assessment solution ARM id tracked by Microsoft.Migrate/migrateProjects."
+        },
+        "projectStatus": {
+          "type": "string",
+          "enum": [
+            "Active",
+            "Inactive"
+          ],
+          "description": "Assessment project status.",
+          "x-ms-enum": {
+            "name": "ProjectStatus",
+            "modelAsString": true
+          }
+        },
+        "customerWorkspaceId": {
+          "type": "string",
+          "description": "The ARM id of service map workspace created by customer."
+        },
+        "customerWorkspaceLocation": {
```

Comment:
why does the user need to provide this?  What happens if they provide an inaccurate value?  If the RP needs the location it should get it from the resource.

Reply:
The customer workspace actually is the Azure service map workspace. The user needs to provide the location so that we know to which service (Azure service bus) endpoint we need to send the call to. If this is provided as inaccurate the call to the service endpoint will fail. The RP is not owning this resource so we can't get its location.